Core/DISET/private/BaseClient.py

```diff
@@ -224,7 +243,18 @@ def _connect( self ):
       transport = gProtocolDict[ self.__URLTuple[0] ][ 'transport' ]( self.__URLTuple[1:3], **self.kwargs )
       retVal = transport.initAsClient()
       if not retVal[ 'OK' ]:
-        return S_ERROR( "Can't connect to %s: %s" % ( self.serviceURL, retVal ) )
+        if self.__retry < 5:
+          url = "%s://%s:%d/%s" % ( self.__URLTuple[0], self.__URLTuple[1], int( self.__URLTuple[2] ), self.__URLTuple[3] )
+          if url not in self.__bannedUrls: 
+            gLogger.info( "URL banned", "%s" % url )
+            self.__bannedUrls += [url]   
+          self.__retry += 1
+          gLogger.info( "Retry connection: ", "%d" % self.__retry )
+          time.sleep( self.__retryDelay )
```

Comment:
Before sleeping, check if there are other urls available. Otherwise every time the connection fails regardless if there are alternatives you will sleep for 2 secs before returning the error.